File: 1997/3/4-3.py
#!/usr/bin/env python

class WoodChuck:
    def __init__(self, name, tooth):
        self.name = name
        self.tooth = tooth

    def __cmp__(self, wc):
        return cmp(self.tooth,wc.tooth)

def bld(x):
    for i in x:
        n, t = i.split()
        yield WoodChuck(n, int(t))

def prnt(x):
    for i in x:
        print i.name, "the woodchuck can chuck", i.tooth * 5, "kilograms of wood."

wc = list(bld(file("prob3_in.txt")))
wc.sort()
prnt(wc)

File: 1997/3/4-3.rb
#!/usr/bin/env ruby

class WoodChuck
  attr_accessor :tooth, :name

  def initialize(name, tooth)
    @name = name
    @tooth = tooth.to_i
  end

  def <=>(wc)
    @tooth <=> wc.tooth
  end
end

list = []
STDIN.readlines.each do |line|
  n,t = line.split()
  list << WoodChuck.new(n, t)
end
list.sort.each do |wc|
  puts "#{wc.name} the woodchuck can chuck #{wc.tooth * 5} kilograms of wood."
end
